[@property] Whitespaces around initial-value should be trimmed

anttijk opened the pull request:
#### 4f0ccda887910822ac9532cc9c50f9953be91d89
<pre>
[@property] Whitespaces around initial-value should be trimmed
<a href="https://bugs.webkit.org/show_bug.cgi?id=249770">https://bugs.webkit.org/show_bug.cgi?id=249770</a>
&lt;rdar://problem/103636334&gt;

Reviewed by Tim Nguyen.

Partially revert <a href="https://commits.webkit.org/258215@main.">https://commits.webkit.org/258215@main.</a> It turns out that WebKit whitespace handling for custom properties
(where whitespace is always trimmed) is correct per <a href="https://drafts.csswg.org/css-syntax-3/#consume-declaration">https://drafts.csswg.org/css-syntax-3/#consume-declaration</a> and this applies to
the `initial-value` property too.

This is tested by imported/w3c/web-platform-tests/css/css-syntax/declarations-trim-whitespace.html which we pass and other engines fail.

Update the @property tests to the spec.

* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-cssom.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/determine-registration-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/determine-registration.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-cssom-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-cssom.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties.html:
* Source/WebCore/css/CSSPropertyRule.cpp:
(WebCore::CSSPropertyRule::cssText const):
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeDeclaration):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::CSSPropertyParser::parseValue):
* Source/WebCore/css/process-css-properties.py:
(PropertiesAndDescriptors.all_descriptor_only):
(Grammar.perform_fixups_for_values_references):
(GenerateCSSPropertyNames):
(PropertiesAndDescriptors.all_preserving_whitespace): Deleted.
(Grammar.preserve_whitespace): Deleted.

Canonical link: <a href="https://commits.webkit.org/258245@main">https://commits.webkit.org/258245@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/caec735e772e0666b2465be45aed07c5e00709c0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101341 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10500 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34401 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110615 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/170887 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105322 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11448 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1353 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93738 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108442 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107124 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8698 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91939 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/35229 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90599 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23346 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/78227 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4119 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/24863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4170 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1281 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10268 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44343 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5673 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5932 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/process-css-properties.py:
```diff
@@ -2663,12 +2653,6 @@ def generate_css_property_names_gperf(self):
 
-            self.generation_context.generate_property_id_switch_function_bool(
-                to=writer,
-                signature="bool CSSProperty::shouldPreserveWhitespace(CSSPropertyID id)",
```
You should get rid of the declaration in `CSSProperty.h` as well.

anttijk replied:
Good point.

nt1m commented on Source/WebCore/css/parser/CSSParserImpl.cpp:
```diff
@@ -1164,8 +1164,7 @@ void CSSParserImpl::consumeDeclaration(CSSParserTokenRange range, StyleRuleType
+    range.consumeWhitespace();
 
     auto declarationValueEnd = range.end();
     bool important = false;
```
Does the next line need reverting as well?

nt1m replied:
(Referring to https://github.com/WebKit/WebKit/commit/11eb340cd2d87f5b32d8b746e829eda7ab2f2d18#diff-8319062f677a6053b9bc0240b714fabe10f2dae2e92d94aa698717de34df4373R1172 )

anttijk replied:
I'll revert it too, not sure if it is right.

anttijk committed:
[@property] Whitespaces around initial-value should be trimmed
https://bugs.webkit.org/show_bug.cgi?id=249770
<rdar://problem/103636334>

Reviewed by Tim Nguyen.

Partially revert https://commits.webkit.org/258215@main. It turns out that WebKit whitespace handling for custom properties
(where whitespace is always trimmed) is correct per https://drafts.csswg.org/css-syntax-3/#consume-declaration and this applies to
the `initial-value` property too.

This is tested by imported/w3c/web-platform-tests/css/css-syntax/declarations-trim-whitespace.html which we pass and other engines fail.

Update the @property tests to the spec.

* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-cssom.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/at-property-shadow.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/determine-registration-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/determine-registration.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-cssom-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-cssom.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/var-reference-registered-properties.html:
* Source/WebCore/css/CSSPropertyRule.cpp:
(WebCore::CSSPropertyRule::cssText const):
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeDeclaration):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::CSSPropertyParser::parseValue):
* Source/WebCore/css/process-css-properties.py:
(PropertiesAndDescriptors.all_descriptor_only):
(Grammar.perform_fixups_for_values_references):
(GenerateCSSPropertyNames):
(PropertiesAndDescriptors.all_preserving_whitespace): Deleted.
(Grammar.preserve_whitespace): Deleted.

Canonical link: https://commits.webkit.org/258245@main
